util/sqldb/sqldb.go

```diff
@@ -0,0 +1,109 @@
+package sqldb
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/upper/db/v4"
+	mysqladp "github.com/upper/db/v4/adapter/mysql"
+	postgresqladp "github.com/upper/db/v4/adapter/postgresql"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/argoproj/argo-workflows/v3/config"
+	"github.com/argoproj/argo-workflows/v3/errors"
+	"github.com/argoproj/argo-workflows/v3/util"
+)
+
+// CreateDBSession creates the dB session
+func CreateDBSession(ctx context.Context, kubectlConfig kubernetes.Interface, namespace string, dbConfig config.DBConfig) (db.Session, error) {
+	if dbConfig.PostgreSQL != nil {
+		return createPostGresDBSession(ctx, kubectlConfig, namespace, dbConfig.PostgreSQL, dbConfig.ConnectionPool)
+	} else if dbConfig.MySQL != nil {
+		return createMySQLDBSession(ctx, kubectlConfig, namespace, dbConfig.MySQL, dbConfig.ConnectionPool)
+	}
+	return nil, fmt.Errorf("no databases are configured")
+}
+
+// createPostGresDBSession creates postgresDB session
+func createPostGresDBSession(ctx context.Context, kubectlConfig kubernetes.Interface, namespace string, cfg *config.PostgreSQLConfig, persistPool *config.ConnectionPool) (db.Session, error) {
```

Comment:
nit, but not your code so its fine. 
Calling it `PostGres` should be illegal.

Reply:
Ha, yeah!